python/tvm/utils/roofline/registry.py

```diff
@@ -15,18 +15,24 @@
 # specific language governing permissions and limitations
 # under the License.
 """Definition of generic functions for estimating peak flops and bandwidth"""
-from typing import Optional
-from ...target import Target, generic_func
-from ...runtime import Device
+from typing import Dict, Optional, Tuple
+
+import numpy as np
+
 from ...rpc.client import RPCSession
+from ...runtime import Device
+from ...target import Target, generic_func
+from ...tir import PrimFunc
 
 
 @generic_func
 def estimate_peak_bandwidth(
+    func: PrimFunc,
```

Comment:
hmm what necessitates this change in interface. It seems to me `func` and `features` (the added inputs) are always used to calculate the added float and str in the return type.

Why not just do the old thing and split up the calls to get the peak statistic, and one to get the actual statistics?

Reply:
You might want to dispatch to different peak measurements based on the features or the primfunc. For example, I could detect if tensorcores could be used with the given primfunc or not.